Functions/src/main/java/com/epam/engx/cleancode/functions/task4/Order.java

```diff
@@ -11,19 +11,22 @@
 
     public Double getPriceOfAvailableProducts() {
         double orderPrice = 0.0;
-        Iterator<Product> iterator = products.iterator();
-        while (iterator.hasNext()) {
-            Product p = iterator.next();
-            if (!p.isAvailable())
-                iterator.remove();
-        }
+        removeUnavailableProducts();
         for (Product p : products)
             orderPrice += p.getProductPrice();
         return orderPrice;
     }
 
-
     public void setProducts(List<Product> products) {
         this.products = products;
     }
+
+    private void removeUnavailableProducts(){
+        Iterator<Product> iterator = products.iterator();
+        while (iterator.hasNext()) {
+            Product p = iterator.next();
+            if (!p.isAvailable())
+                iterator.remove();
```

Comment:
You can change `!p.isAvailable()` to meaningful method. Because `!` hard to see here.